ooniprobe/Engine/Engine.m

```diff
@@ -1,40 +1,90 @@
 #import "Engine.h"
 #import <oonimkall/Oonimkall.h>
-#import <mkall/MKVersion.h>
-#define probeEngineTasks @[@"Telegram", @"Ndt", @"Dash", @"Psiphon", @"Tor", @"Whatsapp", @"FacebookMessenger", @"HttpHeaderFieldManipulation", @"HttpInvalidRequestLine", @"WebConnectivity"]
 
 @implementation Engine
 
-/** getVersionMK returns the version of Measurement Kit we're using */
-+ (NSString*) versionMK {
-    return [MKVersion versionMK];
-}
-
 /** newUUID4 returns the a new UUID4 for this client  */
 + (NSString*) newUUID4 {
     return OonimkallNewUUID4();
 }
 
 /** startExperimentTask starts the experiment described by the provided settings. */
-+ (id<ExperimentTask>) startExperimentTaskWithSettings:(id<ExperimentSettings>)settings error:(NSError **)error {
-    if ([probeEngineTasks containsObject:settings.taskName]) {
-        return [[OONIProbeEngineTaskAdapter alloc]
-                initWithTask:OonimkallStartTask([settings serialization], error)];
-    }
-    return [[MKExperimentTaskAdapter alloc]
-            initWithTask:[MKAsyncTask start:settings.dictionary]];
++ (id<OONIMKTask>) startExperimentTaskWithSettings:(id<OONIMKTaskConfig>)settings error:(NSError **)error{
+    return [[PEMKTask alloc] initWithTask:OonimkallStartTask(settings.serialization,
+                                                             error)];
+}
+
+/** resolveProbeCC returns the probeCC. */
++ (NSString*) resolveProbeCCwithSoftwareName:(NSString*)softwareName
+                             softwareVersion:(NSString*)softwareVersion
+                                     timeout:(long)timeout
+                                       error:(NSError **)error {
+    PESession* session = [[PESession alloc] initWithConfig:
+                          [Engine getDefaultSessionConfigWithSoftwareName:softwareName
+                                                          softwareVersion:softwareVersion
+                                                                   logger:[LoggerNull new]]
+                                                                    error:error];
+    // Updating resources with no timeout because we don't know for sure how much
+    // it will take to download them and choosing a timeout may prevent the operation
+    // to ever complete. (Ideally the user should be able to interrupt the process
+    // and there should be no timeout here.)
+    [session maybeUpdateResources:[session newContext]
+                            error:error];
+    return [session geolocate:[session newContextWithTimeout:timeout]
+                 error:error].country;
```

Comment:
I think we cannot be as compact as we were in Java because here there are errors rather than exceptions:

```suggestion
/** resolveProbeCC returns the probeCC. */
+ (NSString*) resolveProbeCCwithSoftwareName:(NSString*)softwareName
                             softwareVersion:(NSString*)softwareVersion
                                     timeout:(long)timeout
                                       error:(NSError **)userError {
    NSError *___ = nil;
    NSError **error = (userError != nil) ? userError : &___;
    PESession* session = [[PESession alloc] initWithConfig:
                          [Engine getDefaultSessionConfigWithSoftwareName:softwareName
                                                          softwareVersion:softwareVersion
                                                                   logger:[LoggerNull new]]
                                                                    error:error];
    if (error != nil) {
        return nil;
    }
    // Updating resources with no timeout because we don't know for sure how much
    // it will take to download them and choosing a timeout may prevent the operation
    // to ever complete. (Ideally the user should be able to interrupt the process
    // and there should be no timeout here.)
    [session maybeUpdateResources:[session newContext] error:error];
    if (error != nil) {
        return nil;
    }
    OONIGeolocateResults *r = [session geolocate:[session newContextWithTimeout:timeout] error:error];
    if (error != nil) {
        return nil;
    }
    return r.country;
```